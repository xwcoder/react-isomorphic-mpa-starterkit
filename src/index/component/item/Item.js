import React from 'react'
import formatTime from '../../../lib/formatTime'

export default function Item ({ data }) {
  const {
    url,
    pubTime,
    title
  } = data

  return (
    <li className="list-item">
      <a href={url}>{title}</a>
      <span className="tag-time">{formatTime(pubTime, 'yyyy-MM-dd hh:mm:ss')}</span>
    </li>
  )
}
