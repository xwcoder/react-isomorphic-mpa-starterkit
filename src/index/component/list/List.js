import React from 'react'
import Item from '../item'

export default function List ({ list }) {

  const items = list.map(data => <Item key={data.docId} data={data} />)

  return (
    <ul className="list">{items}</ul>
  )
}
