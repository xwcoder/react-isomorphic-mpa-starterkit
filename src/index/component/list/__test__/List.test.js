import React from 'react'
import { shallow } from 'enzyme'
import List from '../List'
import Item from '../../item'

const list = [
  {
    title: '还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！',
    url: 'http://kuaibao.qq.com/s/20180202C0H7YJ00',
    pubTime: 1522816928000,
    docId: 20,
    summary: '还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！'
  },
  {
    title: '还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！',
    url: 'http://kuaibao.qq.com/s/20180202A0A93800',
    pubTime: 1522809128000,
    docId: 7,
    summary: '还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！'
  },
  {
    title: '这个国家美女成灾愁嫁人 你想去吗？',
    url: 'http://kuaibao.qq.com/s/20180202A0A93800',
    pubTime: 1522807988000,
    docId: 22,
    summary: '这个国家美女成灾愁嫁人 你想去吗？这个国家美女成灾愁嫁人 你想去吗？'
  }
]


describe('index/component/List', () => {

  it('list with three data', () => {
    const wrapper = shallow(<List list={list} />)
    expect(wrapper.find(Item)).toHaveLength(3)
    expect(wrapper.find(Item).at(1).key()).toEqual(`${list[1].docId}`)
    expect(wrapper).toMatchSnapshot()
  })
})
