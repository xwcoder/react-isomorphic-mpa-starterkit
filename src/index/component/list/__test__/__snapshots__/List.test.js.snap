// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/component/List list with three data 1`] = `
<ul
  className="list"
>
  <Item
    data={
      Object {
        "docId": 20,
        "pubTime": 1522816928000,
        "summary": "还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！",
        "title": "还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！",
        "url": "http://kuaibao.qq.com/s/20180202C0H7YJ00",
      }
    }
    key="20"
  />
  <Item
    data={
      Object {
        "docId": 7,
        "pubTime": 1522809128000,
        "summary": "还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！",
        "title": "还在说“别人家”的包子吗？蒸包子的8个技巧，轻松成为“别人家”的包子！",
        "url": "http://kuaibao.qq.com/s/20180202A0A93800",
      }
    }
    key="7"
  />
  <Item
    data={
      Object {
        "docId": 22,
        "pubTime": 1522807988000,
        "summary": "这个国家美女成灾愁嫁人 你想去吗？这个国家美女成灾愁嫁人 你想去吗？",
        "title": "这个国家美女成灾愁嫁人 你想去吗？",
        "url": "http://kuaibao.qq.com/s/20180202A0A93800",
      }
    }
    key="22"
  />
</ul>
`;
